packages/kit/src/runtime/client/fetcher.js

```diff
@@ -1,9 +1,10 @@
-import { DEV } from 'esm-env';
+import { BROWSER, DEV } from 'esm-env';
 import { hash } from '../hash.js';
 
 let loading = 0;
 
-export const native_fetch = window.fetch;
+/** @type {typeof fetch} */
+export const native_fetch = BROWSER ? window.fetch : /** @type {any} */ (() => {});
```

Comment:
I wouldn't expect we'd need a check for `BROWSER` inside this file because I'd think that `client/fetcher` should only be loaded in the client to begin with

Reply:
Oh, I guess `app/navigation` is imported on both the client and server and so this is to avoid pulling extra stuff into the server. It mostly shouldn't matter on the server, but I guess some serverless lambda stuff have size limits?

Perhaps we could solve this instead by having `app/client-navigation` and `app/server-navigation` and then returning the appropriate one where the Vite alias is setup? That way we only have a single browser vs server check instead of one for each function we define

Reply:
I looked into implementing this a bit and I'm not sure it's worthwhile / totally doable. The problem is that `forms.js` also needs to be split into two separate files then, and [it doesn't sound like the "load a different file depending on client or server isn't easily possible in Vite](https://vitejs.dev/config/#conditional-config)? At the very least, I'm not quite sure how to accomplish it.

Reply:
I'm not sure if there's something particular you were referring to in that link. I was thinking that you could create an alias for `$app/navigation` on the line below and return a different value depending on whether it's the client or server build:

https://github.com/sveltejs/kit/blob/0679801fb9e673928c29cf03c5d3566753510ab6/packages/kit/src/exports/vite/index.js#L260

I don't think you need to bother splitting `forms.js` since it only has a single check. I think it's more the navigation file where the biggest benefit is since you have to repeat the check so many times there

Reply:
I don't think we should mess around with aliases. It makes it harder to keep things up to date, makes types more complicated, and in general just adds moving parts. Much simpler to use `BROWSER` in the few places it's necessary.